wgpu/src/lib.rs

```diff
@@ -1936,12 +1958,29 @@ impl Instance {
     /// - On macOS/Metal: will panic if not called on the main thread.
     /// - On web: will panic if the `raw_window_handle` does not properly refer to a
     ///   canvas element.
-    pub unsafe fn create_surface<
-        W: raw_window_handle::HasWindowHandle + raw_window_handle::HasDisplayHandle,
+    pub fn create_surface<
+        'window,
+        W: 'window + HasWindowHandle + HasDisplayHandle + WasmNotSend + WasmNotSync,
```

Comment:
Nit: put the lifetime last, use WgpuSurfaceRequirement and move to where clause

Reply:
Why make `WgpuSurfaceRequirement` public if it works with existing types already?
I thought the only reason why `WgpuSurfaceRequirement` exists at all is because it's required to store the window in a `Box<dyn T>`, which doesn't allow multiple non-auto traits.

(in the meantime I will just go ahead and still apply the suggestion)

Reply:
Oh yeah, you're totally right, this can just be internal as it's only needed to physically construct the dyn we want - actually, we don't need to access the window handle at all from the dyn. We can just use Option<Box <dyn Any>> and get rid of this trait entirely.

Reply:
> We can just use Option<Box > and get rid of this trait entirely.

Did you have an idea what to store in the `Box`? I couldn't remove the trait entirely because we still need *any* trait **and** `Send + Sync`.

I was thinking of introducing a new `WasmNotSendSync` trait, which might shorten a lot of lines in the codebase as well as a follow-up?

Reply:
Sure, that sounds reasonable.

Note that `Send + Sync` are special in dyn - you can do `dyn Trait + Send + Sync` fine. You can't do `dyn Trait + WasmNotSend + WasmNotSync` as they aren't the "special" ones :)